x-pack/plugins/enterprise_search/public/applications/shared/role_mapping/attribute_selector.tsx

```diff
@@ -0,0 +1,187 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import {
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiPanel,
+  EuiSelect,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+
+import {
+  ANY_AUTH_PROVIDER,
+  ANY_AUTH_PROVIDER_OPTION_LABEL,
+  AUTH_ANY_PROVIDER_LABEL,
+  AUTH_INDIVIDUAL_PROVIDER_LABEL,
+  ATTRIBUTE_SELECTOR_TITLE,
+  AUTH_PROVIDER_LABEL,
+  EXTERNAL_ATTRIBUTE_LABEL,
+  ATTRIBUTE_VALUE_LABEL,
+} from './constants';
+
+interface IAttributeSelectorProps {
+  attributeName: string;
+  attributeValue?: string;
+  attributes: string[];
+  selectedAuthProviders?: string[];
+  availableAuthProviders?: string[];
+  elasticsearchRoles: string[];
+  disabled: boolean;
+  multipleAuthProvidersConfig: boolean;
+  handleAttributeSelectorChange(value: string, elasticsearchRole: string): void;
+  handleAttributeValueChange(value: string): void;
+  handleAuthProviderChange?(value: string[]): void;
+}
+
+interface AttributeExamples {
+  username: string;
+  email: string;
+  metadata: string;
+}
+
+interface ParentOption extends EuiComboBoxOptionOption<string> {
+  label: string;
+  options: ChildOption[];
+}
+
+interface ChildOption extends EuiComboBoxOptionOption<string> {
+  value: string;
+  label: string;
+}
+
+const attributeValueExamples = {
+  username: 'elastic,*_system',
+  email: 'user@example.com,*@example.org',
+  metadata: '{"_reserved": true}',
+};
+
+const getAuthProviderOptions = (availableAuthProviders: string[]) => {
+  return [
+    {
+      label: AUTH_ANY_PROVIDER_LABEL,
+      options: [{ value: ANY_AUTH_PROVIDER, label: ANY_AUTH_PROVIDER_OPTION_LABEL }],
+    },
+    {
+      label: AUTH_INDIVIDUAL_PROVIDER_LABEL,
+      options: availableAuthProviders.map((authProvider) => ({
+        value: authProvider,
+        label: authProvider,
+      })),
+    },
+  ];
+};
+
+const getSelectedOptions = (selectedAuthProviders: string[], availableAuthProviders: string[]) => {
+  const groupedOptions: ParentOption[] = getAuthProviderOptions(availableAuthProviders);
+  const childOptions: ChildOption[] = [];
+  const options = groupedOptions.reduce((acc, n) => [...acc, ...n.options], childOptions);
+  return options.filter((o) => o.value && selectedAuthProviders.includes(o.value));
+};
+
+export const AttributeSelector: React.FC<IAttributeSelectorProps> = ({
+  attributeName,
+  attributeValue = '',
+  attributes,
+  availableAuthProviders,
+  selectedAuthProviders = [ANY_AUTH_PROVIDER],
+  elasticsearchRoles,
+  disabled,
+  multipleAuthProvidersConfig,
+  handleAttributeSelectorChange,
+  handleAttributeValueChange,
+  handleAuthProviderChange = () => null,
+}) => {
+  return (
+    <EuiPanel
+      data-test-subj="attributeSelector"
```

Comment:
I think it's somewhat standard to use upper-cased names for data-test-subj in Kibana. I know we haven't been consistent with that, but I just thought I'd mention it.

`AttributeSelector`

Reply:
There's some other instances of this in the code, if you're going to update this you might want to look for and fix other instances also, like `authProviderSelect`.